x-pack/plugins/siem/public/components/error_toast/index.tsx

```diff
@@ -22,40 +21,33 @@ interface ReduxProps {
 }
 
 interface DispatchProps {
-  addError?: ActionCreator<{ id: string; title: string; message: string }>;
-  removeError?: ActionCreator<{ id: string }>;
+  removeError: ActionCreator<{ id: string }>;
 }
 
 type Props = OwnProps & ReduxProps & DispatchProps;
 
-const ErrorToastComponent = pure<Props>(({ toastLifeTimeMs = 10000, errors = [], removeError }) =>
-  globalListFromToasts(errorsToToasts(errors), removeError!, toastLifeTimeMs)
-);
-
-export const globalListFromToasts = (
-  toasts: Toast[],
-  removeError: ActionCreator<{ id: string }>,
-  toastLifeTimeMs: number
-) =>
-  toasts.length !== 0 ? (
-    <EuiGlobalToastList
-      toasts={toasts}
-      dismissToast={({ id }) => removeError({ id })}
-      toastLifeTimeMs={toastLifeTimeMs}
-    />
-  ) : null;
-
-export const errorsToToasts = (errors: appModel.Error[]): Toast[] =>
-  errors.map(({ id, title, message }) => {
-    const toast: Toast = {
-      id,
-      title,
-      color: 'danger',
-      iconType: 'alert',
-      text: <p>{message}</p>,
-    };
-    return toast;
+const ErrorToastComponent = ({ toastLifeTimeMs = 5000, errors = [], removeError }: Props) => {
+  const [{ toasts }, dispatchToaster] = useStateToaster();
+  useEffect(() => {
+    errors.forEach(({ id, title, message }) => {
+      if (toasts.filter(toast => toast.id === id).length === 0) {
```

Comment:
I would use the semantic `some` boolean to see if the value does not exist.

```ts
if (!toasts.some(toast => toast.id === id)) {
```

or `find`. Filter seems incorrect semantically